packages/x-components/src/x-modules/facets-next/components/lists/__tests__/filters-injection.mixin.spec.ts

```diff
@@ -0,0 +1,183 @@
+import { Filter } from '@empathyco/x-types-next';
+import { mount, Wrapper } from '@vue/test-utils';
+import Vue, { ComponentOptions } from 'vue';
+import { Component, Prop } from 'vue-property-decorator';
+import {
+  createNextHierarchicalFilter,
+  createNextSimpleFilter
+} from '../../../../../__stubs__/filters-stubs.factory';
+import { XProvide } from '../../../../../components/decorators/injection.decorators';
+import FiltersInjectionMixin from '../filters-injection.mixin';
+
+@Component({
+  template: `
+    <div>
+    <slot/>
+    </div>`
+})
+class Provider extends Vue {
+  @Prop()
+  @XProvide('filters')
+  public injectedData!: any;
+}
+
+const Filters: ComponentOptions<Vue> = {
+  mixins: [FiltersInjectionMixin],
+  template: `
+    <div>
+    <li v-for="filter of renderedFilters" :key="filter.id">{{ filter.id }}</li>
+    </div>
+  `
+};
+
+describe('Test filters injection mixin', () => {
+  function mountComponent({
+    propFilters,
+    injectedFilters,
+    parentId
+  }: Partial<{
+    propFilters: Filter[];
+    injectedFilters: Filter[];
+    parentId: Filter['id'];
+  }> = {}): TestFilterInjectionAPI {
+    const wrapper = mount(
+      {
+        template: `
+          <Provider :injectedData="injectedFilters">
+          <Filters :filters="propFilters" :parentId="parentId"/>
+          </Provider>
+        `,
+        components: { Provider, Filters }
+      },
+      {
+        data() {
+          return {
+            propFilters,
+            injectedFilters,
+            parentId
+          };
+        }
+      }
+    );
+
+    return {
+      wrapper,
+      getRenderedFilterIds(): string[] {
+        return wrapper.findAll('li').wrappers.map(element => element.text());
+      },
+      async setPropFilters(propFilters: Filter[]) {
+        await wrapper.setData({ propFilters });
```

Comment:
There is an error in the IDE regarding this method:

![image](https://user-images.githubusercontent.com/3502075/128813369-47c09e34-54ae-4c0c-8ebf-5f250bb89748.png)

`setData(format: string, data: string): void;`

Reply:
I don't know why the Webstorm take that method from other code instead the wrapper of vue test utils.
It is a fail of webstorm I think, not from our code.